### PR TITLE
Store last root check point along wit its slot

### PR DESCRIPTION
**Pull request by jmozah**
This PR adds the DB functions to save and get the last checkpoint along with its slot.

PS: These needs to be used in appropriate place to store the read the checkpoints @potuz

### CODE REVIEW

File: beacon-chain/db/kv/validated_checkpoint.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	ctx, span := trace.StartSpan(ctx, "BeaconDB.LastValidatedCheckpoint")
+	defer span.End()
+
+	var lastChkPoint [32]byte
+	var slot types.Slot
+	err := s.db.View(func(tx *bolt.Tx) error {
+		bkt := tx.Bucket(lastValidatedCheckpoint)
+		val := bkt.Get([]byte("lastChkPoint"))
+		if len(val) != 40 {
+			return errors.New("invalid checkpoint point")
```
Comment (potuz) [inline beacon-chain/db/kv/validated_checkpoint.go:24]: We are trying to not use errors.New but rather create a named error in errors.go

File: beacon-chain/db/kv/validated_checkpoint.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	"encoding/binary"
+
+	"github.com/pkg/errors"
+	types "github.com/prysmaticlabs/eth2-types"
+	"github.com/prysmaticlabs/prysm/encoding/bytesutil"
+	bolt "go.etcd.io/bbolt"
+	"go.opencensus.io/trace"
+)
+
+func (s *Store) LastValidatedCheckpoint(ctx context.Context) ([32]byte, types.Slot, error) {
```
Comment (potuz) [inline beacon-chain/db/kv/validated_checkpoint.go:14]: This is not exported in the declaration

Comment (jmozah) [reply beacon-chain/db/kv/validated_checkpoint.go:14]: I assume that this function will be used outside DB pkg for getting the latest validated checkpoint and its slot.

PS: i forgot to export the save function... i will export that too

**Comment by jmozah** [comment]
this is superceded by https://github.com/prysmaticlabs/prysm/pull/10351. so closing

**Pull request closed without merging**